gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/AbstractJWTFilter.java

```diff
@@ -170,7 +170,13 @@ protected void configureExpectedParameters(FilterConfig filterConfig) {
   protected boolean tokenIsStillValid(JWT jwtToken) {
     Date expires;
     if (tokenStateService != null) {
-      expires = new Date(tokenStateService.getTokenExpiration(jwtToken.toString()));
+      long timestamp = 0;
+      try {
+        timestamp = tokenStateService.getTokenExpiration(jwtToken.toString());
+      } catch (Exception e) {
```

Comment:
Do we not have a specific exception coming from getTokenExpiration? or tokenStateService?

Reply:
I've been debating the need for an UnknownTokenException, and your comment may have just convinced me to implement it.

Reply:
I would rather change the method signatures in a separate commit, so I've created https://issues.apache.org/jira/browse/KNOX-2230 to address this concern.